### PR TITLE
Add Snaps documentation for MetaMask Extension 12.6 updates

### LINKED ISSUE #1673: Document Snaps functionality to be able to update context
Can now update context in snap_updateInterface

### LINKED ISSUE #1701: Document Snaps Avatar component
Document `<Avatar>` in custom UI

### LINKED ISSUE #1702: Document min, max, and step props for Snaps Input Component
`<Input>` now accepts optional min, max, and step props

### LINKED ISSUE #1703: Document size prop for Snaps Heading component
`<Heading>` now accepts optional size prop

### LINKED ISSUE #1704: Document metamask: scheme URLs for Snaps Links
Snaps can now link to their own homepage and the MetaMask main page in custom UI links. 

TO DO: get examples of each link.

### LINKED ISSUE #1705: Update Snaps Row Component to include Link as a possible child
`<Row>` can now contain `<Link>`

### LINKED ISSUE #1706: Update Snaps Link component to include Address as a possible child
`<Link>` can now contain `<Address>`

### LINKED ISSUE #1711: Update Snaps address component per recent changes
`<Address>` now supports non-EVM addresses. We should also mention that we do not do any checksum validation for non-EVM addresses.

### PR DESCRIPTION
**Pull request by Montoya**
# Description

Multiple Snaps docs updates to coincide with the release of MetaMask Extension 12.6

## Issue(s) fixed

Fixes #1673

Fixes #1701 

Fixes #1711 

Fixes #1702 

Fixes #1703 

Fixes #1704 

Fixes #1705 

Fixes #1706 

## Checklist

Complete this checklist before merging your PR:

- [ ] If this PR contains a major change to the documentation content, I have added an entry to the top of the ["What's new?"](https://github.com/MetaMask/metamask-docs/blob/main/docs/whats-new.md) page. TO-DO
- [ ] The proposed changes have been reviewed and approved by a member of the documentation team.

### CODE REVIEW

File: snaps/features/custom-ui/index.md [hunk 1]
```diff
@@ -738,8 +796,17 @@ Outputs a clickable link.
...
+  `https:`, `mailto:`, and `metamask:`. `http:` is not allowed.
+- `children`: `Array<string | Bold | Italic | Address>` - The link text, 
+  or an [`Address`](#address).
+
+#### `metamask:` URLs
+
+A Snap can link to the following screens using the `metamask:` scheme: 
+
+- `metamask://client/` - Leads to the main screen of MetaMask 
+- `metamask://snap/home/` - Leads to the Snap's [home page](../custom-ui/home-pages.md)
```
Comment (FrederikBolding) [inline snaps/features/custom-ui/index.md:809]: Not sure if it is clear that `snap` should be a Snap ID.

Comment (Montoya) [reply snaps/features/custom-ui/index.md:809]: Oh whoops, let me correct that

Comment (Montoya) [reply snaps/features/custom-ui/index.md]: Fixed.

**Review by FrederikBolding — APPROVED** [review]

**Review by alexandratran — APPROVED** [review]
Added some minor edits. LGTM.

**Pull request merged**